source/slang/slang.cpp

```diff
@@ -587,8 +587,18 @@ void FrontEndCompileRequest::parseTranslationUnit(
     TranslationUnitRequest* translationUnit)
 {
     IncludeHandlerImpl includeHandler;
-    includeHandler.linkage = getLinkage();
-    includeHandler.searchDirectories = &searchDirectories;
+
+    auto linkage = getLinkage();
+
+    // TODO(JS): NOTE! Here we are using the searchDirectories on the linkage. This is because
+    // currently the API only allows the setting search paths on linkage.
+    // 
+    // Here we should probably be using the searchDirectories on the FrontEndCompileRequest.
+    // If searchDirectories.parent pointed to the one in the Linkage would mean linkage paths
+    // would be checked too (after those on the FrontEndCompileRequest). 
```

Comment:
I know we already talked about this on another channel, but changing it so that whenever we create a `FrontEndCompileRequest` (possibly in its constructor if it has one...) we set the `searchDirectories.parent` pointer would actually be a smaller change than this one.
I'm fine either way, since the user-observable behavior is fixed either way, and we can revisit the implementation later when we start figuring out how to expose the distinct search paths.